src/main/java/com/google/sps/servlets/EditRecipeServlet.java

```diff
@@ -67,17 +78,65 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         steps = FormHelper.separateByNewlines(stepsResponse);
       }
 
+      // recipe is private and stays private
       recipeEntity.setProperty("name", name);
       recipeEntity.setProperty("tags", tags);
       recipeEntity.setProperty("description", description);
       recipeEntity.setProperty("ingredients", ingredients);
       recipeEntity.setProperty("steps", steps);
-      recipeEntity.setProperty("imageBlobKey", (String) recipeEntity.getProperty("imageBlobKey"));
+      recipeEntity.setProperty("imageBlobKey", imageBlobKey);
+
+      if (privacy.equals("public")) {
+        recipeEntity.setProperty("published", true);
+      }
+      if (privacy.equals("private")) {
+        recipeEntity.setProperty("published", false);
+      }
+
+      // recipe was private and is made public
+      if (!published && privacy.equals("public")) {
```

Comment:
What if a recipe is public and was made private?

Reply:
starts at line 126